server/controllers/v1/post.ts

```diff
@@ -105,6 +106,81 @@ export default class PostController {
     };
   }
 
+  async searchForPosts(
+    userID: string,
+    query: string,
+    limit: number,
+    offset: number
+  ): Promise<{
+    status: number;
+    data: { result?: PostUserPreview[]; count: number; total: number };
+  }> {
+    // await db.sequelize.query(`DROP TABLE IF EXISTS "PostSearches";`);
+    // await db.sequelize.query(
+    //   `CREATE TEMP TABLE "PostSearches" AS
+    //   SELECT *, (
+    //     setweight(to_tsvector(coalesce("title",'')), 'A') ||
+    //     setweight(to_tsvector(coalesce("location",'')), 'C') ||
+    //     setweight(to_tsvector(coalesce("body",'')), 'D')
+    //   ) as "tsvector"
+    //   FROM "Posts";`
+    // );
+    const data = await Promise.all([
+      db.sequelize.query(
+        `SELECT 
+          "Post"."id", 
+          "Post"."body", 
+          "Post"."title", 
+          "Post"."createdAt", 
+          "Post"."thumbnail", 
+          (SELECT COUNT(*) FROM "UserPostLikes" AS "Likes" 
+            WHERE "Likes"."postID" = "Post"."id") AS "likeCount", 
+          (SELECT COUNT(*) FROM "UserPostLikes" AS "Likes" 
+            WHERE "Likes"."postID" = "Post"."id" AND "Likes"."userID" = $1) AS "doesUserLike", 
+          json_build_object(
+            'firstName', "User"."firstName", 
+            'lastName', "User"."lastName",
+            'id', "User"."id"
+          ) AS "User",
+          (
+            setweight(to_tsvector(coalesce("Post"."title",'')), 'A') || 
+            setweight(to_tsvector(coalesce("Post"."location",'')), 'C') || 
+            setweight(to_tsvector(coalesce("Post"."body",'')), 'D')
+          ) AS "tsvector",
+          ts_rank_cd((
+            setweight(to_tsvector(coalesce("Post"."title",'')), 'A') || 
+            setweight(to_tsvector(coalesce("Post"."location",'')), 'C') || 
+            setweight(to_tsvector(coalesce("Post"."body",'')), 'D')
+          ), "query", 1|4) AS "rank" 
+        FROM "Posts" AS "Post" 
+        LEFT OUTER JOIN "Users" AS "User" ON "Post"."UserId" = "User"."id" 
+        CROSS JOIN to_tsquery($2) AS "query" 
+        WHERE "query" @@ (
```

Comment:
Is `"query"` the user input sent from the client? Shouldn't we sanitize this since raw sequelize queries don't have SQLI protection?

Reply:
prepared statements should be enough to circumvent sqli